### PR TITLE
Matrix

### PR DESCRIPTION
**Pull request by grahamc**
##### Checklist

- [ ] Tested changes against a test repository
- [ ] Added or updated relevant documentation (leave unchecked if not applicable)
- [ ] (If this PR is for a release) Updated README to point to the new tag (leave unchecked if not applicable)

### CODE REVIEW

**Comment by grahamc** [comment]
Includes #97 and #95

File: src/index.ts [hunk 1]
```diff
@@ -224,12 +224,15 @@ class MagicNixCacheAction extends DetSysAction {
       diagnosticEndpoint,
       "--nix-conf",
       nixConfPath,
+      "--use-gha-cache",
+      useGhaCache,
+      "--use-flakehub",
+      useFlakeHub,
```
[the remark this answers is not in the document]
Comment (colemickens) [reply src/index.ts:230]: Oh, it's literally the next section, never mind, I see how they moved.

File: .github/workflows/ci.yml [hunk 2]
```diff
@@ -93,20 +99,41 @@ jobs:
             runner: "ubuntu-22.04"
     steps:
       - uses: actions/checkout@v4
-        if: github.event_name == 'merge_group'
       - name: Install Nix on ${{ matrix.systems.nix-system }} system
-        if: github.event_name == 'merge_group'
         uses: DeterminateSystems/nix-installer-action@main
         with:
-          flakehub: true
+          _internal-obliterate-actions-id-token-request-variables: ${{ matrix.id-token == 'none' }}
```
Comment (colemickens) [inline .github/workflows/ci.yml:105]: what's this for? `_internal-obliterate...` ?

Comment (grahamc) [reply .github/workflows/ci.yml:105]: I want those actions to not have the ACTIONS_ID_TOKEN_REQUEST_* environment variables, so they can't create an ID token and log in to flakehub. This simulates not having the `id-token: write` permission. I wanted to parameterize that permission in the matrix, but permissions can't be variable.

**Review by colemickens — APPROVED** [review]
awesome, very slick, definitely increases my confidence about things.

**Pull request merged**